src/main/java/uk/gov/hmcts/darts/audio/service/impl/AudioServiceImpl.java

```diff
@@ -245,6 +249,23 @@ public void setIsArchived(List<AudioMetadata> audioMetadata, Integer hearingId)
         }
     }
 
+    /*
+    Set the isAvailable flag if the media is available in the unstructured datastore.
+     */
+    @Override
+    public void setIsAvailable(List<AudioMetadata> audioMetadataList) {
+        List<Integer> mediaIdList = audioMetadataList.stream().map(AudioMetadata::getId).toList();
+        ObjectRecordStatusEntity storedStatus = objectRecordStatusRepository.getReferenceById(STORED.getId());
+        ExternalLocationTypeEntity unstructuredLocationType = externalLocationTypeRepository.getReferenceById(UNSTRUCTURED.getId());
+        List<ExternalObjectDirectoryEntity> storedInUnstructuredList = externalObjectDirectoryRepository.findByInMediaIdStatusAndType(mediaIdList, storedStatus,
```

Comment:
Not sure if it would be worth just returning the IDs from the query, to save having to do the map below?

Reply:
i've changed. i thought by returning the ExternalObjectDirectoryEntity it would be more reusable, but maybe the query is too specific to be of use to anyone else.

Reply:
Yeah I wasn't sure for the same reason, but happy either way 👍